ui/app/AppLayouts/Profile/popups/AddShowTokenModal.qml

```diff
@@ -0,0 +1,183 @@
+        if (addressInput.text !== "" && !Utils.isAddress(addressInput.text)) {
+            //% "This needs to be a valid address"
+            validationError = qsTrId("this-needs-to-be-a-valid-address");
+        }
+        return validationError === ""
+    }
+
+    property var getTokenDetails: Backpressure.debounce(popup, 500, function (tokenAddress){
+        popup.walletStore.walletTokensModule.getTokenDetails(tokenAddress)
+    });
+
+    function onKeyReleased(){
+        validationError = "";
+        if (!validate() || addressInput.text === "") {
+            return;
+        }
+        Qt.callLater(getTokenDetails, addressInput.text)
+    }
+
+    Item {
+        Connections {
+            target: popup.walletStore.walletTokensModule
+            onTokenDetailsWereResolved: {
+                const jsonObj = JSON.parse(tokenDetails)
+                if (jsonObj.error) {
+                    validationError = jsonObj.error
+                    return
+                }
+                if (jsonObj.name === "" && jsonObj.symbol === "" && jsonObj.decimals === "") {
+                    //% "Invalid ERC20 address"
+                    validationError = qsTrId("invalid-erc20-address")
+                    return;
+                }
+
+                if (addressInput.text.toLowerCase() === jsonObj.address.toLowerCase()) {
+                    symbolInput.text = jsonObj.symbol;
+                    decimalsInput.text = jsonObj.decimals;
+                    nameInput.text = jsonObj.name;
+                }
+            }
+        }
+    }
+
+    Input {
+        id: addressInput
+        readOnly: !editable
+        textField.maximumLength: 42
+        //% "Enter contract address..."
+        placeholderText: qsTrId("enter-contract-address...")
+        //% "Contract address"
+        label: qsTrId("contract-address")
+        validationError: popup.validationError
+        Keys.onReleased: onKeyReleased()
+    }
+
+    Input {
+        id: nameInput
+        readOnly: !editable
+        anchors.top: addressInput.bottom
+        anchors.topMargin: marginBetweenInputs
+        //% "The name of your token..."
+        placeholderText: qsTrId("the-name-of-your-token...")
+        //% "Name"
+        label: qsTrId("name")
+    }
+
+    Input {
+        id: symbolInput
+        readOnly: !editable
+        //% "ABC"
+        placeholderText: qsTrId("abc")
+        //% "Symbol"
+        label: qsTrId("symbol")
+        anchors.top: nameInput.bottom
+        anchors.topMargin: marginBetweenInputs
+        anchors.left: parent.left
+        anchors.right: undefined
+        width: parent.width / 2 - 20
+    }
+
+    Input {
+        id: decimalsInput
+        readOnly: !editable
+        placeholderText: "18"
+        //% "Decimals"
+        label: qsTrId("decimals")
+        text: "18"
+        anchors.top: nameInput.bottom
+        anchors.topMargin: marginBetweenInputs
+        anchors.right: parent.right
+        anchors.left: undefined
```

Comment:
we can remove this in case we want to make it undefined

Reply:
If I remove it, the 2 inputs overlap :(